### PR TITLE
Add vSphere IPI destroy

### PR DESCRIPTION
**Pull request by jcpowermac**
- Modify clustermetadata to include VSphere
- Add additional fields for auth to vsphere Metadata struct
  that is required for accessing vSphere vCenter
  to destroy cluster infrastructure objects
- vSphere Destroy
  - Create clients for both soap and rest
  - Finds the vCenter objects attached to the Tag
  - Retieve the ManagedObjects for both Folder and VirtualMachine
  - Confirm that exactly one Folder is available
  - Remove VirtualMachines that are a child of Folder
  - Remove Folder if empty

### CODE REVIEW

**Comment by jcpowermac** [comment]
Need to update Gopkg (will do in another commit)

**Comment by jcpowermac** [comment]
@patrickdillon 
@mtnbikenc 
when you have a moment if I can get an initial review.

**Comment by jcpowermac** [comment]
Looks like:
https://github.com/openshift/machine-api-operator/blob/master/go.mod#L28
is using govmomi version v0.21.0 we might want to use that version as well.

File: pkg/types/vsphere/client.go [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+	"net/url"
+	"time"
+
+	"github.com/vmware/govmomi"
+	"github.com/vmware/govmomi/vapi/rest"
+	"github.com/vmware/govmomi/vim25"
+	"github.com/vmware/govmomi/vim25/soap"
+)
+
+func CreateVSphereClients(ctx context.Context, vcenter, username, password string) (*vim25.Client, *rest.Client, error) {
```
Comment (jcpowermac) [inline pkg/types/vsphere/client.go]: @jstuever ^

File: pkg/destroy/vsphere/vsphere.go [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+
+	return attached, nil
+}
+
+// Run is the entrypoint to start the uninstall process.
+func (o *ClusterUninstaller) Run() error {
+	var folderList []types.ManagedObjectReference
+	var virtualMachineList []types.ManagedObjectReference
+
+	//Based on the vSphere vCenter Tag (InfraID or ClusterID - TBD)
```
Comment (patrickdillon) [inline pkg/destroy/vsphere/vsphere.go]: This comment can be removed (or updated as we are using `InfraID`)

**Comment by patrickdillon** [comment]
Logging changes look good to me.
Before change there was no output:
```
$ ./openshift-install create cluster --dir vsphere
$
```

Now deleted resources are output:
```
$ ./openshift-install destroy cluster --dir vsphere
INFO Destroyed                                     VirtualMachine=test-cluster-v2mww-master-1
INFO Destroyed                                     VirtualMachine=test-cluster-v2mww-master-0
INFO Destroyed                                     VirtualMachine=test-cluster-v2mww-master-2
INFO Destroyed                                     VirtualMachine=test-cluster-v2mww-bootstrap
INFO Destroyed                                     Folder=test-cluster-v2mww
$
```
For reference, here is log with `--log-level debug`
```
$ ./openshift-install destroy cluster --dir vsphere --log-level debug
DEBUG OpenShift Installer unreleased-master-2383-gaade1eb75561332885c102eac3ebd2af3c6e8b20 
DEBUG Built from commit aade1eb75561332885c102eac3ebd2af3c6e8b20 
DEBUG find attached objects on tag                 
DEBUG find Folder objects                          
DEBUG find VirtualMachine objects                  
DEBUG delete VirtualMachines                       
DEBUG Powered off                                   VirtualMachine=test-cluster-sp8d9-master-0
INFO Destroyed                                     VirtualMachine=test-cluster-sp8d9-master-0
DEBUG Powered off                                   VirtualMachine=test-cluster-sp8d9-master-2
INFO Destroyed                                     VirtualMachine=test-cluster-sp8d9-master-2
DEBUG Powered off                                   VirtualMachine=test-cluster-sp8d9-master-1
INFO Destroyed                                     VirtualMachine=test-cluster-sp8d9-master-1
DEBUG Powered off                                   VirtualMachine=test-cluster-sp8d9-bootstrap
INFO Destroyed                                     VirtualMachine=test-cluster-sp8d9-bootstrap
DEBUG find Folder objects                          
DEBUG delete Folder                                
INFO Destroyed                                     Folder=test-cluster-sp8d9
DEBUG Purging asset "Terraform Variables" from disk 
DEBUG Purging asset "Kubeconfig Admin Client" from disk 
DEBUG Purging asset "Kubeadmin Password" from disk 
DEBUG Purging asset "Certificate (journal-gatewayd)" from disk 
DEBUG Purging asset "Metadata" from disk           
DEBUG Purging asset "Cluster" from disk            
$ 
```

/lgtm

**Comment by jcpowermac** [comment]
/hold
e2e-aws-upgrade broken

**Comment by jcpowermac** [comment]
/hold
e2e-aws-upgrade broken

**Pull request merged**